### PR TITLE
CLI-507 Add e2e tests for sonar-context-augmentation

### PR DESCRIPTION
**Pull request by mpaladin**
## Summary

Adds an offline e2e suite under `tests/e2e/context/` that exercises the **real** `sonar-context-augmentation` binary — download, PGP signature verification, tar extraction, skill rendering, post-update refresh, and passthrough — without touching SonarQube/Cloud. The only network reach is `binaries.sonarsource.com` for the archive and detached signature.

The trigger path uses pre-seeded `state.json` (stale `config.cliVersion` + stale-version skill record) so `runPostUpdateActions()` fires and walks the real install chain, bypassing the entitlement check and `cag init` that would otherwise require a live SQ Cloud connection.

### Coverage

| File | Tests | What it proves |
|---|---|---|
| `cag-offline.test.ts` | 11 | Initial install (binary path, signature, `--version`, state bump, `SKILL.md` written, `sonar context --help` passthrough) + simulated self-update reinstall |
| `cag-edge-cases.test.ts` | 5 | Deleted projectRoot is skipped, global skill is skipped, multi-skill refresh in one pass, no-op when cliVersion is current, stale-version binary cleanup |
| `cag-passthrough.test.ts` | 3 | Unauthenticated non-help action emits the CLI's \"Not authenticated\" error, bare \`sonar context\` falls through to CAG's help, child non-zero exit propagates to the parent |
| `cag-copilot.test.ts` | 2 | Post-update renders `.github/skills/sonar-context-augmentation/SKILL.md` for a `copilot-cli` skill record |
| `_helpers.ts` | – | Shared `buildSkillExtension` / `seedState` / `findRecordedCagSkill` scaffolding |

**Total: 21 tests in ~6.3s locally.**

### Not in scope

The two \`integrate\` guard rails I considered (CAG skipped on SonarQube Server, CAG skipped with \`--global\`) are already covered by integration tests at `tests/integration/specs/integrate/context-augmentation.test.ts:454,570`. Since those paths return *before* the binary would be downloaded, an e2e variant exercises no real-binary code that integration doesn't already cover.

Live SonarQube Cloud coverage (entitlement check, \`cag init\`, real \`sonar context <action>\`) is deferred — those require Vault secrets and a CAG-entitled test org.

----
## Summary by Gitar

- **Test enhancements:**
  - Added `tests/e2e/context/cag-integrate.test.ts` to verify that Context Augmentation is correctly skipped for SonarQube Server connections and disabled cloud entitlements.
  - Updated `CLAUDE.md` to document the new `cag-integrate.test.ts` E2E test suite coverage.

<sub>This will update automatically on new commits.</sub>

### CODE REVIEW

**Comment by mpaladin** [comment]
@sonar-review-alpha review

File: tests/e2e/context/cag-integrate.test.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+const DEFAULT_TIMEOUT_MS = 60_000;
+const INTEGRATE_TIMEOUT_MS = 30_000;
+
+setDefaultTimeout(DEFAULT_TIMEOUT_MS);
+
+const PROJECT_KEY = 'cag-e2e-project';
+const ORG_KEY = 'cag-e2e-org';
+const TOKEN = 'cag-e2e-token';
+
+function findRecordedCagSkill(state: CliState): SkillExtension | undefined {
```
Comment (kirill-knize-sonarsource) [inline tests/e2e/context/cag-integrate.test.ts]: Looks like this guy belongs to the `_helpers.ts`. There's a similar function defined there.

**Review by kirill-knize-sonarsource — APPROVED** [review]
LGTM
Minor comment

File: tests/e2e/context/cag-integrate.test.ts (hunk 1 above)
Comment (mpaladin) [reply tests/e2e/context/cag-integrate.test.ts]: @kirill-knize-sonarsource fixed in https://github.com/SonarSource/sonarqube-cli/pull/317/changes/f838faf804ca8e034526a25863d77937799e4b44.
